File: LMIS/LMIS.Web/Modules/ReaderManage/LibraryCards/LibraryCardsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.ReaderManage.Columns;

[ColumnsScript("ReaderManage.LibraryCards")]
[BasedOnRow(typeof(LibraryCardsRow), CheckNames = true)]
public class LibraryCardsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public long LibraryCardId { get; set; }
    [EditLink]
    public string LibraryCardNo { get; set; }
    public string Username { get; set; }
    public string LevelCardLevelName { get; set; }
    public short Status { get; set; }
    public DateTime CreateTime { get; set; }
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/ReaderManage/LibrarycardsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LibraryCardsRow } from "./LibraryCardsRow";

export interface LibraryCardsColumns {
    LibraryCardId: Column<LibraryCardsRow>;
    LibraryCardNo: Column<LibraryCardsRow>;
    Username: Column<LibraryCardsRow>;
    LevelCardLevelName: Column<LibraryCardsRow>;
    Status: Column<LibraryCardsRow>;
    CreateTime: Column<LibraryCardsRow>;
    UpdateTime: Column<LibraryCardsRow>;
}

export class LibraryCardsColumns extends ColumnsBase<LibraryCardsRow> {
    static readonly columnsKey = 'ReaderManage.LibraryCards';
    static readonly Fields = fieldsProxy<LibraryCardsColumns>();
}